### PR TITLE
Console subscriber page hooked to bff

### PR DESCRIPTION
**Pull request by Brackleycassinga**

### CODE REVIEW

File: interfaces/console/ui/molecules/SubscriberDetails/index.tsx [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+            <Tab label="History" />
+          </Tabs>
+        </Box>
+
+        <Box sx={{ pl: 2 }}>
+          <Typography
+            component="div"
+            role="tabpanel"
+            hidden={selectedTab !== 0}
+          >
```
Comment (salman-codewar) [inline interfaces/console/ui/molecules/SubscriberDetails/index.tsx:196]: We can use `Box` here, rather `Typography`

File: nodes/ukamaOS/distro/system/configd/.cproject [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (salman-codewar) [inline nodes/ukamaOS/distro/system/configd/.cproject:1]: Why we have ukamaOS changes, any idea?

**Review by salman-codewar — CHANGES_REQUESTED** [review]

File: interfaces/console/ui/molecules/AddSubscriber/Step1.tsx [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+                      <Typography
+                        variant="body2"
+                        sx={{ color: colors.black70 }}
+                      >
+                        {` (${pSimCount || 0} left)`}
+                      </Typography>
+                    </Stack>
+                  </Stack>
+                </Paper>
+              </Grid>
```
[the remark this answers is not in the document]
Comment (Brackleycassinga) [reply interfaces/console/ui/molecules/AddSubscriber/Step1.tsx:151]: Addressed

File: interfaces/console/ui/molecules/AddSubscriber/index.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        return 'Enter the ICCID for the SIM you have assigned to the subscriber, and select their data plan. Please ensure the ICCID is correct, because it cannot be undone once assigned.';
+      case 3:
+        return selectedSimType == 'eSim '
+          ? `You have successfully added ${name} as a subscriber to your network, and an ${selectedSimType} installation invitation has been sent out to them. If they would rather install their eSIM now, have them scan the QR code below.`
+          : `You have successfully added ${name} as a subscriber to your network, and ${selectedSimType} installation instructions have been sent out to them. `;
+      default:
+        return 'Add subscribers to your network.';
+    }
+  };
+
```
[the remark this answers is not in the document]
Comment (Brackleycassinga) [reply interfaces/console/ui/molecules/AddSubscriber/index.tsx:118]: Addressed

File: interfaces/console/ui/molecules/SubscriberDetails/index.tsx [hunk 5]
```diff
@@ -0,0 +1,457 @@
...
+        <Box sx={{ p: 2 }}>
+          <Tabs
+            value={selectedTab}
+            onChange={handleTabChange}
+            aria-label="tabs menu"
+          >
+            <Tab label="Information" />
+            <Tab label="Data Usage" />
+            <Tab label="SIMs" />
+            <Tab label="History" />
```
[the remark this answers is not in the document]
Comment (Brackleycassinga) [reply interfaces/console/ui/molecules/SubscriberDetails/index.tsx:187]: Addressed

File: interfaces/console/ui/molecules/SubscriberDetails/index.tsx (hunk 1 above)
Comment (Brackleycassinga) [reply interfaces/console/ui/molecules/SubscriberDetails/index.tsx:196]: Addressed

File: systems/console-bff/subscriber/resolver/types.ts [hunk 6]
```diff
@@ -114,40 +114,40 @@ export class SubscribersAPIResDto {
...
   network_id: string;
 
   @Field()
   org_id: string;
 
-  @Field()
-  @IsPhoneNumber()
-  phone: string;
+  @Field({ nullable: true })
+  // @IsPhoneNumber()
```
Comment (salman-codewar) [inline systems/console-bff/subscriber/resolver/types.ts:146]: we can remove it.

File: systems/console-bff/sim/resolver/types.ts [hunk 7]
```diff
@@ -167,24 +223,106 @@ export class DeleteSimInputDto {
...
   simId: string;
 }
+// {
+//   "package_id": "{{PackageUUID}}",
+//   "sim_id": "{{SimUUID}}",
+//   "start_date": {
+//     "nanos": 0,
+//     "seconds": 0
+//   }
+// }
```
Comment (salman-codewar) [inline systems/console-bff/sim/resolver/types.ts:233]: We can remove these comments too

File: systems/console-bff/package.json [hunk 8]
```diff
@@ -44,6 +44,7 @@
     "express": "^4.18.2",
     "express-winston": "^4.2.0",
     "graphql": "latest",
+    "graphql-iso-date": "^3.6.1",
```
Comment (salman-codewar) [inline systems/console-bff/package.json:47]: Let's remove this package

**Review by salman-codewar — CHANGES_REQUESTED** [review]

**Review by salman-codewar — APPROVED** [review]

**Comment by salman-codewar** [comment]
Looks good man 👍

**Pull request merged**